cmake/kokkos_arch.cmake

```diff
@@ -680,7 +680,7 @@ endif()
 
 if(NOT KOKKOS_COMPILE_LANGUAGE STREQUAL CUDA)
   if(KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE)
-    compiler_specific_flags(Clang -fcuda-rdc NVIDIA --relocatable-device-code=true)
+    compiler_specific_flags(Clang -fgpu-rdc --offload-new-driver NVIDIA --relocatable-device-code=true)
```

Comment:
I could not find documentation about `--offload-new-driver`.
Why do we need it?

Reply:
See slide 13, https://www.openmp.org/wp-content/uploads/OpenMP-Offloading-in-LLVM-15.pdf. This is not just about OpenMP offloading, but enables OpenMP target offload and HIP/CUDA interoperability. Without `--offload-new-driver` you get the errors like `undefined reference to `__cudaRegisterLinkedBinary__nv_50a56434682ff3e9` in issue #2421.

Reply:
Also see https://www.youtube.com/watch?v=4NnzymmQe7k for more context.

Reply:
Btw, I did try to also add `--offload-link` to the linker flags, but that produces "unused command-line" option warnings. That's why I left that out. Despite this, all unit tests seem to pass.

Reply:
offload-new-driver is an optimization that does among other things substantially reduces file IO.  It also bundles global variable initializations into a single HIP or Cuda call, which can make a marked difference in program startup (as in 5+ mins while it iterates over the symbol table setting each global one at a time). For SNL apps all this put together is rather substantial.

Reply:
Finally found a meaningful documentation page: https://openmp.llvm.org/CommandLineArgumentReference.html